pkg/operator/periodicbackupcontroller/periodicbackupcontroller.go

```
@@ -272,3 +374,215 @@ func newCronJob() (*batchv1.CronJob, error) {
+							SecurityContext: &corev1.SecurityContext{
+								Privileged: ptr.To(true),
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	return &ds
+}
+
+func getEtcdEndpoints(ctx context.Context, client kubernetes.Interface) ([]string, error) {
+	etcdEndPointsCM, err := client.CoreV1().ConfigMaps(operatorclient.TargetNamespace).Get(ctx, etcdEndpointConfigMapName, v1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	var endpoints []string
+	for _, v := range etcdEndPointsCM.Data {
+		ep := v + etcdClientPort
+		endpoints = append(endpoints, ep)
+	}
+
+	slices.Sort(endpoints)
+	return endpoints, nil
+}
+
+func constructBackupServerArgs(cr backupv1alpha1.Backup, endpoints string) []string {
+	backupConfig := backuphelpers.NewDisabledBackupConfig()
+	backupConfig.SetBackupSpec(&cr.Spec.EtcdBackupSpec)
+	argList := backupConfig.ArgList()
+
+	argList = append(argList, fmt.Sprintf("--endpoints=%s", endpoints))
+	argList = append(argList, fmt.Sprintf("--backupPath=%s", backuprestore.BackupVolume))
+
+	return argList
+}
+
+func checkBackupServerPodsStatus(podLister clientv1.PodLister) ([]string, error) {
+	backupPods, err := podLister.List(labels.Set{backupDSLabelKey: backupDSLabelValue}.AsSelector())
+	if err != nil {
+		return nil, fmt.Errorf("PeriodicBackupController could not list etcd pods: %w", err)
+	}
+
+	var terminationReasons []string
+	for _, p := range backupPods {
+		if p.Status.Phase == corev1.PodFailed {
+			for _, cStatus := range p.Status.ContainerStatuses {
+				if cStatus.Name == etcdBackupServerContainerName {
+					// TODO we can also try different cStatus.State.Terminated.ExitCode
+					terminationReasons = append(terminationReasons, fmt.Sprintf("container %s within pod %s has been terminated: %s", etcdBackupServerContainerName, p.Name, cStatus.State.Terminated.Message))
+				}
+			}
+		}
+	}
+
+	return terminationReasons, nil
+}
+
+func etcdBackupServerDSDiffers(l, r appv1.DaemonSetSpec) bool {
```

This ignores a lot of errors, imo it should return any found error (so that it would be reported) and treat errors as "DS are different"

---

so this func just compares the bytes to find if any diff exist, what error could I report ? 🤔

---

There might be an error on marshaling but yeah, its a nitpicking

---

yeah make sense 👍🏽

---

it complicates the code flow unnecessarily, I believe its not vital, wdyt ?